src/spectral.ts

```diff
@@ -52,7 +55,18 @@ export class Spectral {
 
   constructor(protected readonly opts?: IConstructorOpts) {
     this._computeFingerprint = memoize(opts?.computeFingerprint ?? defaultComputeResultFingerprint);
-    this._resolver = opts?.resolver ?? new Resolver();
+
+    if (opts?.proxyUri) {
+      const ProxyAgent = require('proxy-agent');
```

Comment:
I'm a bit on edge with importing `proxy-agent` dynamically here, and in `spectral.ts` in general.
This is very likely to cause major troubles for folks that use Spectral in a browser, as their bundlers are likely to:
- include the package in the final bundle, since it's not loading asynchronously via `import`, therefore vastly increasing the size (https://bundlephobia.com/result?p=proxy-agent@3.1.1)
- this may lead to some compatibility issues if the code of that package is evaluated, and they will need to be forced shim certain node.js modules.

How about changing `proxyUri` in `IConstructorOpts` `agent?: Agent` to `IConstructorOpts` and assigning the agent here?

Reply:
To throw out some other options: would using the `browsers` field in package.json work as described here?
- https://github.com/defunctzombie/package-browser-field-spec#ignore-a-module
- https://github.com/webpack/webpack/issues/8826#issuecomment-469642169

Or using `eval` as Sindre Sorhus suggests:
- https://twitter.com/kamilogorek/status/1102272038411137025

Passing an Agent to Spectral is simple but feels a little less natural. Though I'm happy to change to that if you prefer it.

Reply:
@AndrewBell I'm fine with throwing eval, as this is likely to be the most compatible solution.
`browser` field will depend on the bundler you use, while `eval` is unlikely to be actually recognized by most of (if not all) bundlers, as they usually look for a require call expression and do not verify eval calls.
So, let's go with `eval`.

Reply:
Last but not least, could you leave a comment explaining why we decided to go with `eval`? This might seem confusing for others in future.

Reply:
@P0lip, I just thought of one scenario where `eval` will still be an issue: if you are using a bundler for non-browser context. For example, we bundle and minify lambda code. Seems like you really want a different entrypoint for node to avoid all these problems. One for another day perhaps...

Reply:
>  if you are using a bundler for non-browser context.

True, however this is rather less likely, hopefully.

> Seems like you really want a different entrypoint for node to avoid all these problems. One for another day perhaps...

Yeah, I have this thought in mind.
CLI code could also be better separated, perhaps to a different package.
We'll certainly revisit it at some point when we are in a better spot with other features.